<!--

Record a test stream from the camera, using color and depth compression, into a ROS bag for later playback
roslaunch realsense_ros_camera record_compressed_bag.launch bag_path:=my_value

--> 
<launch>
  <arg name="fisheye_width"       default="640" />
  <arg name="fisheye_height"      default="480" />
  <arg name="enable_fisheye"      default="true" />
  <arg name="depth_width"         default="320" />
  <arg name="depth_height"        default="240" />
  <arg name="enable_depth"        default="true" />
  <arg name="color_width"         default="640" />
  <arg name="color_height"        default="480" />
  <arg name="enable_color"        default="true" />

  <arg name="bag_path" default="$(env HOME)/test.bag" />
  <arg name="bag_topics" default="/camera/color/image_raw/compressed /camera/color/camera_info /camera/depth/image_raw/compressedDepth /camera/depth/camera_info /camera/fisheye/camera_info /camera/fisheye/image_raw /camera/accel/sample /camera/gyro/sample /camera/accel/imu_info /camera/gyro/imu_info /camera/extrinsics/fisheye2imu /camera/extrinsics/fisheye2depth"/>

  <node pkg="rosbag" type="record" name="rosbag" args="-O $(arg bag_path) $(arg bag_topics)"/>
  <node pkg="nodelet" type="nodelet" name="realsense_ros_camera" args="standalone realsense_ros_camera/NodeletCamera" output="screen">
    <param name="fisheye_width"            type="int"  value="$(arg fisheye_width)" />
    <param name="fisheye_height"           type="int"  value="$(arg fisheye_height)" />
    <param name="enable_fisheye"           type="bool" value="$(arg enable_fisheye)" />
    <param name="depth_width"              type="int"  value="$(arg depth_width)" />
    <param name="depth_height"             type="int"  value="$(arg depth_height)" />
    <param name="enable_depth"             type="bool" value="$(arg enable_depth)" />
    <param name="color_width"              type="int"  value="$(arg color_width)" />
    <param name="color_height"             type="int"  value="$(arg color_height)" />
    <param name="enable_color"             type="bool" value="$(arg enable_color)" />
  </node>
</launch>
